plugins/identity-shiro/src/main/java/org/opensearch/identity/shiro/ShiroIdentityPlugin.java

```diff
@@ -138,7 +140,7 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
         }
     }
 
-    public PluginSubject getPluginSubject(Plugin plugin) {
-        return new ShiroPluginSubject(threadPool);
+    public Client getRunAsClient(Plugin plugin) {
```

Comment:
(Style) I prefer `createRunAsClient()` but I'm ok with consistency if this is a similar pattern elsewhere.

Reply:
I don't think `create` is the right work here since it won't be instantiating a new client under the hood. For its implemtation, the client returned from this method will be a FilterClient that wraps the nodeClient that's passed to this plugin in createComponents.